tests/appsec/test_automated_login_events.py

```diff
@@ -37,10 +45,16 @@ class Test_Login_Events:
     BASIC_AUTH_INVALID_PASSWORD_HEADER = "Basic dGVzdDoxMjM0NQ=="  # base64(test:12345)
 
     def setup_login_pii_success(self):
-        self.r_pii_success = [
-            weblog.post("/login?auth=local", data={"username": self.USER, "password": self.PASSWORD}),
-            weblog.get("/login?auth=basic", headers={"Authorization": self.BASIC_AUTH_USER_HEADER}),
-        ]
+        if context.library == "ruby":
```

Comment:
Could you add methods to the class: 

```python
class Test_Login_Events:
    """ Test login success/failure use cases """

    @property
    def username_key(self):
       """ <Explanation why is it different> """
       return "user[username]" if "rails" in context.weblog_variant else "username"

    # same for password_key
```

Then, setup method will be all the same : 

```python
                weblog.post("/login?auth=local", data={self.username_key: self.USER, self.password_key: self.PASSWORD}),
```